src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1040,6 +1040,19 @@ internal void OnAdditionalDocumentTextUpdatedOnDisk(DocumentId documentId)
             this.OnAdditionalDocumentTextLoaderChanged(documentId, vsDoc.Loader);
         }
 
+        internal void OnIsComplete(ProjectId projectId, bool succeeded)
+        {
+            // this can't be null
+            var state = CurrentSolution.GetProjectState(projectId);
```

Comment:
Why is this still groveling into ProjectState, which should be internal to the core Workspace type?